core/src/toga/platform.py

```diff
@@ -73,43 +73,50 @@ def get_platform_factory() -> ModuleType:
             # (https://github.com/chaquo/chaquopy/issues/890), so include the original
             # exception message in case the backend does exist but throws a
             # ModuleNotFoundError from one of its internal imports.
-            raise RuntimeError(
+            msg = (
                 f"The backend specified by TOGA_BACKEND ({backend_value!r}) could "
                 f"not be loaded ({e}). It should be one of: {toga_backends_values}."
             )
+            raise RuntimeError(msg)
```

Comment:
Pre-emptive protection against Ruff's [EM102 rule](https://docs.astral.sh/ruff/rules/f-string-in-exception/). Even though we're not using Ruff, I think the advice is good. Since I was reformatting a bunch of f-strings because of a listing change, I figured I'd get ahead of a future problem.

Reply:
I don't think Ruff enables that rule by default, and I don't think we should enable it. We spend more time looking at code than we do a tracebacks, so it seems a poor tradeoff to make the traceback more readable at the expense of making the code less readable, and also longer.

Reply:
I'm not sure I agree it makes the code less readable; but given it's an optional part of this PR anyway, I'm happy to leave that litigation to later. I'll revert that part of the change.